src/main/java/com/yahoo/bullet/storm/BulletStormConfig.java

```diff
@@ -38,30 +46,255 @@
     public static final String JOIN_BOLT_CPU_LOAD = "bullet.topology.join.bolt.cpu.load";
     public static final String JOIN_BOLT_MEMORY_ON_HEAP_LOAD = "bullet.topology.join.bolt.memory.on.heap.load";
     public static final String JOIN_BOLT_MEMORY_OFF_HEAP_LOAD = "bullet.topology.join.bolt.memory.off.heap.load";
-    public static final String JOIN_BOLT_QUERY_TICK_TIMEOUT = "bullet.topology.join.bolt.query.tick.timeout";
     public static final String RESULT_BOLT_PARALLELISM = "bullet.topology.result.bolt.parallelism";
     public static final String RESULT_BOLT_CPU_LOAD = "bullet.topology.result.bolt.cpu.load";
     public static final String RESULT_BOLT_MEMORY_ON_HEAP_LOAD = "bullet.topology.result.bolt.memory.on.heap.load";
     public static final String RESULT_BOLT_MEMORY_OFF_HEAP_LOAD = "bullet.topology.result.bolt.memory.off.heap.load";
-    public static final String TICK_INTERVAL_SECS = "bullet.topology.tick.interval.secs";
+    public static final String LOOP_BOLT_PARALLELISM = "bullet.topology.loop.bolt.parallelism";
+    public static final String LOOP_BOLT_CPU_LOAD = "bullet.topology.loop.bolt.cpu.load";
+    public static final String LOOP_BOLT_MEMORY_ON_HEAP_LOAD = "bullet.topology.loop.bolt.memory.on.heap.load";
+    public static final String LOOP_BOLT_MEMORY_OFF_HEAP_LOAD = "bullet.topology.loop.bolt.memory.off.heap.load";
+    public static final String TICK_SPOUT_INTERVAL = "bullet.topology.tick.spout.interval";
+    public static final String JOIN_BOLT_QUERY_TICK_TIMEOUT = "bullet.topology.join.bolt.query.tick.timeout";
+    public static final String JOIN_BOLT_WINDOW_TICK_TIMEOUT = "bullet.topology.join.bolt.window.tick.timeout";
+    public static final String LOOP_BOLT_PUBSUB_OVERRIDES = "bullet.topology.loop.bolt.pubsub.overrides";
+
+    // Defaults
+
+    public static final String DEFAULT_TOPOLOGY_NAME = "bullet-topology";
+    public static final boolean DEFAULT_TOPOLOGY_METRICS_ENABLE = false;
+    public static final boolean DEFAULT_TOPOLOGY_METRICS_BUILT_IN_ENABLE = false;
+    public static final Map<String, Number> DEFAULT_TOPOLOGY_METRICS_BUILT_IN_EMIT_INTERVAL_MAPPING = new HashMap<>();
+    public static final String DEFAULT_BUILT_IN_METRICS_INTERVAL_KEY = "default";
+    static {
+        DEFAULT_TOPOLOGY_METRICS_BUILT_IN_EMIT_INTERVAL_MAPPING.put("bullet_active_queries", 10L);
+        DEFAULT_TOPOLOGY_METRICS_BUILT_IN_EMIT_INTERVAL_MAPPING.put(DEFAULT_BUILT_IN_METRICS_INTERVAL_KEY, 10L);
+    }
+    public static final List<String> DEFAULT_TOPOLOGY_METRICS_CLASSES = new ArrayList<>();
+    static {
+        DEFAULT_TOPOLOGY_METRICS_CLASSES.add(SigarLoggingMetricsConsumer.class.getName());
```

Comment:
put SigarLoggingMetricsConsumer.class.getName() as param when new ArrayList?

Reply:
There is no such constructor so I will have to create a new Collection to pass in.